tests/Remove-DbaAgentJob.Tests.ps1

```diff
@@ -0,0 +1,77 @@
+$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
+. "$PSScriptRoot\constants.ps1"
+
+Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
+    Context "Validate parameters" {
+        $paramCount = 7
+        $defaultParamCount = 13
+        [object[]]$params = (Get-ChildItem function:\Remove-DbaAgentJob).Parameters.Keys
+        $knownParameters = 'SqlInstance', 'SqlCredential', 'Job', 'KeepHistory', 'KeepUnusedSchedule', 'Mode', 'EnableException'
+        It "Should contain our specific parameters" {
+            ( (Compare-Object -ReferenceObject $knownParameters -DifferenceObject $params -IncludeEqual | Where-Object SideIndicator -eq "==").Count ) | Should Be $paramCount
+        }
+        It "Should only contain $paramCount parameters" {
+            $params.Count - $defaultParamCount | Should Be $paramCount
+        }
+    }
+}
+Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
+    Context "Command removes jobs" {
+        BeforeAll {
+            $null = New-DbaAgentSchedule -SqlInstance $script:instance2 -Schedule dbatoolsci_daily -FrequencyType Daily -FrequencyInterval Everyday -Force
```

Comment:
This just hit me but might be a good idea for time to just create the connection to your test instance using `Connect-DbaInstance`...can help shave off some time since you are calling multiple commands. Just have that connection and pass it to each command.

Would also give you access to the `Query()` method when you need to execute SQL statements.

Reply:
You would call the connect within each context block to prevent conflicts.